gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/PalantirJavaFormatProviderPlugin.java

```diff
@@ -18,36 +18,87 @@
         Preconditions.checkState(
                 rootProject == rootProject.getRootProject(),
                 "May only apply com.palantir.java-format-provider to the root project");
 
-        Configuration configuration = rootProject.getConfigurations().create(CONFIGURATION_NAME, conf -> {
-            conf.setDescription("Internal configuration for resolving the palantir-java-format implementation");
-            conf.setVisible(false);
-            conf.setCanBeConsumed(false);
-
-            conf.defaultDependencies(deps -> {
-                deps.add(rootProject
-                        .getDependencies()
-                        .create(ImmutableMap.of(
-                                "group", "com.palantir.javaformat",
-                                "name", "palantir-java-format",
-                                "version",
-                                        JavaFormatExtension.class.getPackage().getImplementationVersion())));
-            });
+        String implementationVersion = JavaFormatExtension.class.getPackage().getImplementationVersion();
+
+        Configuration configuration = rootProject.getConfigurations().create(CONFIGURATION_NAME);
+        configuration.setDescription("Internal configuration for resolving the palantir-java-format implementation");
+        configuration.setVisible(false);
+        configuration.setCanBeConsumed(false);
+        configuration.setCanBeResolved(true);
+        configuration.defaultDependencies(deps -> {
+            deps.add(rootProject
+                    .getDependencies()
+                    .create(ImmutableMap.of(
+                            "group",
+                            "com.palantir.javaformat",
+                            "name",
+                            "palantir-java-format",
+                            "version",
+                            implementationVersion)));
+        });
+
+        Configuration nativeConfiguration = rootProject.getConfigurations().create(NATIVE_CONFIGURATION_NAME);
+        nativeConfiguration.setDescription(
+                "Internal configuration for resolving the palantir-java-format implementation");
+        nativeConfiguration.setVisible(false);
+        nativeConfiguration.setCanBeConsumed(false);
+        nativeConfiguration.setCanBeResolved(true);
+        nativeConfiguration.defaultDependencies(deps -> {
+            deps.add(rootProject
+                    .getDependencies()
+                    .create(ImmutableMap.of(
+                            "group",
+                            "com.palantir.javaformat",
+                            "name",
+                            "palantir-java-format-native",
+                            "version",
+                            implementationVersion,
+                            "classifier",
+                            String.format("nativeImage-%s", getCurrentOs()),
+                            "ext",
+                            "exe")));
```

Comment:
This is pretty hard to read - I'd either use the string version `String.format("com.palantir.javaformat:palantir-java-format-native:%s:%s@%s")` or at least `builder()` variant for the map so each part lines up